natlib/svcd.c

```diff
@@ -30,6 +31,63 @@ static const LUA_REG_TYPE svcd_meta_map[] =
 // Maintainer: Michael Andersen <michael@steelcode.com>
 /////////////////////////////////////////////////////////////
 
+//Lua: storm.n.svcd_advert_received(pay, srcip, srcport)
+//Dispatch an advertisment
+
+int svcd_advert_received(lua_State *L) 
+{
+	char *srcip = luaL_checkstring(L, 2);
+        printf("Service advertisment %s\n", srcip);
+	
+	//check if pay is nil, don't proceed if true
+	int nil_pay=lua_isnil(L,1);
+        if(nil_pay==1)
+	{
+ 		return 0;
+	}
+
+	char *pay = luaL_checkstring(L, 1);
+	int srcport = luaL_checknumber(L, 3);
+	lua_pushlightfunction(L, libmsgpack_mp_unpack);      
+       	lua_pushstring(L,pay); 
+	lua_call(L,1,1);	
+ 	lua_pushvalue(L, -1);
```

Comment:
Maybe you should add a comment explaining this. lua_pushvalue(L, -1) appears to just duplicate the top of the stack, and it's unclear from reading your code why you need that.